### PR TITLE
Add Closed loop BZ#2173199

### PR DESCRIPTION
**Pull request by shubhamsg199**
Adding closed loop for https://bugzilla.redhat.com/show_bug.cgi?id=2173199

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_bootdisk.py -k test_positive_bootdisk_download_https

File: robottelo/cli/bootdisk.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @classmethod
+    def generic(cls, options=None):
+        """Associate a resource"""
+        cls.command_sub = 'generic'
+        return cls.execute(cls._construct_command(options), output_format='json')
+
+    @classmethod
+    def host(cls, options=None):
+        """Add subscription"""
```
Comment (adarshdubey-star) [inline robottelo/cli/bootdisk.py:33]: Is this the correct description of the method?

File: robottelo/cli/bootdisk.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        cls.command_sub = 'host'
+        return cls.execute(cls._construct_command(options), output_format='json')
+
+    @classmethod
+    def subnet(cls, options=None):
+        """Override product content defaults"""
```
Comment (adarshdubey-star) [inline robottelo/cli/bootdisk.py:39]: More descriptive doc strings..

File: robottelo/cli/bootdisk.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class Bootdisk(Base):
+    """Manipulates Bootdisk."""
+
+    command_base = 'bootdisk'
+
+    @classmethod
+    def generic(cls, options=None):
+        """Associate a resource"""
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply robottelo/cli/bootdisk.py]: Updated

File: tests/foreman/cli/test_bootdisk.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+:CaseLevel: System
+
+:CaseComponent: BootdiskPlugin
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: Critical
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply tests/foreman/cli/test_bootdisk.py:15]: Updated to High

File: robottelo/cli/bootdisk.py (hunk 1 above)
Comment (shubhamsg199) [reply robottelo/cli/bootdisk.py]: Updated

File: tests/foreman/cli/test_bootdisk.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+"""
+import pytest
+from fauxfactory import gen_mac
+from fauxfactory import gen_string
+
+from robottelo.cli.factory import make_medium
+from robottelo.config import settings
+from robottelo.constants import HTTPS_MEDIUM_URL
+
+
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_bootdisk.py:27]: Marking on_prem_provisioning might be appropriate for this test

Comment (shubhamsg199) [reply tests/foreman/cli/test_bootdisk.py:27]: Not needed as we are just checking if the full host bootdisk is downloaded properly for the created host. Bootdisk provisioning is in the plan which I will add in future.

File: tests/foreman/cli/test_bootdisk.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    :BZ: 2173199
+
+    :customerscenario: true
+
+    :parametrized: yes
+    """
+
+    macaddress = gen_mac(multicast=False)
+    capsule = module_target_sat.nailgun_smart_proxy
+    capsule.location = [module_location]
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply tests/foreman/cli/test_bootdisk.py:60]: Right, not needed. Removing it, Thanks.

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_bootdisk.py -k test_positive_bootdisk_download_https

File: tests/foreman/cli/test_bootdisk.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    # create medium with https url
+    media = make_medium(
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply tests/foreman/cli/test_bootdisk.py]: Updated

File: robottelo/constants/__init__.py [hunk 8]
```diff
@@ -199,6 +199,8 @@ class Colored(Box):
 
 INSTALL_MEDIUM_URL = "http://mirror.fakeos.org/%s/$major.$minor/os/$arch"
 
+HTTPS_MEDIUM_URL = "https://partha.fedorapeople.org/test-repos/kickstart-zoo"
```
[the remark this answers is not in the document]
Comment (shubhamsg199) [reply robottelo/constants/__init__.py:202]: Yes, for now, I'll prefer to keep it in constants :)

File: robottelo/host_helpers/cli_factory.py [hunk 9]
```diff
@@ -1157,3 +1158,25 @@ def setup_cdn_and_custom_repos_content(
...
+        args = {
+            'location-ids': None,
+            'locations': None,
+            'name': gen_alphanumeric(6),
+            'operatingsystem-ids': None,
+            'operatingsystems': None,
+            'organization-ids': None,
+            'organizations': None,
+            'os-family': None,
+            'path': 'http://{}'.format(gen_string('alpha', 6)),
```
Comment (Gauravtalreja1) [inline robottelo/host_helpers/cli_factory.py:1179]:
```suggestion
            'path': 'http://{}'.format(gen_string('alpha', 6)),
```
```suggestion
            'path': 'http://{gen_string('alpha', 6)}',
```

File: robottelo/host_helpers/cli_factory.py [hunk 10]
```diff
@@ -1157,3 +1158,25 @@ def setup_cdn_and_custom_repos_content(
             data['lce'] = lce
 
         return data
+
+    def make_medium(self, options=None):
+        """Creates a Medium
+
```
Comment (Gauravtalreja1) [inline robottelo/host_helpers/cli_factory.py:1164]: it seems dict for make_medium is already defined in ENTITY_FIELDS, so might not need this func

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_bootdisk.py -k test_positive_bootdisk_download_https

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_bootdisk.py -k test_positive_bootdisk_download_https

File: robottelo/host_helpers/cli_factory.py (hunk 10 above)
Comment (shubhamsg199) [reply robottelo/host_helpers/cli_factory.py]: Right, updated.

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**